state/protocol/state.go

```diff
@@ -37,25 +37,40 @@ type State interface {
 	AtBlockID(blockID flow.Identifier) Snapshot
 }
 
-type MutableState interface {
+type FollowerState interface {
```

Comment:
```suggestion
// FollowerState is a mutable protocol state used by nodes following main consensus (ie. non-consensus nodes).
// All blocks must have a certifying QC when being added to the state to guarantee they are valid,
// so there is a one-block lag between block production and incorporation into the FollowerState.
// However, since all blocks are certified upon insertion, they are immediately processable by other components. 
type FollowerState interface {
```